src/librustc/middle/cstore.rs

```diff
@@ -67,6 +67,8 @@ pub struct CrateSource {
 
 #[derive(RustcEncodable, RustcDecodable, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Debug)]
 pub enum DepKind {
+    /// A dependency that is only used for its macros, none of which are visible from other crates.
+    UnexportedMacrosOnly,
```

Comment:
`UnexportedMacrosOnly` are only included in the metadata as placeholders (the `CrateNum` of a dependency is computed from its position in the dependency list) -- I'll add a comment.

Does that answer your question? (not sure how monomorphised functions are relevant)